src/Utilities/Utilities.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Pinterest for WooCommerce Utilities.
+ *
+ * @package     Pinterest_For_WooCommerce/Classes/
+ * @since       x.x.x
+ */
+
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\Pinterest\Utilities;
+
+use \Error;
+
+/**
+ * Utilities class.
+ *
+ * @since x.x.x
+ */
+class Utilities {
+
+	const PINTEREST_ACCOUNT_CONNECTION_TIMESTAMP = PINTEREST_FOR_WOOCOMMERCE_PREFIX . '-account-connection-timestamp';
+
+	/**
+	 * Set the account connection timestamp.
+	 *
+	 * @since x.x.x
+	 * @throws Error Wrong value provided for the timestamp.
+	 * @param int $timestamp Timestamp to set as the account connection moment.
+	 * @return int
+	 */
+	public static function set_account_connection_timestamp( $timestamp = null ): int {
+		if ( is_null( $timestamp ) ) {
+			$timestamp = time();
+		}
+
+		if ( ! is_int( $timestamp ) ) {
+			throw new Error( 'Account connection timestamp needs to be an integer value.' );
+		}
+
+		add_option( self::PINTEREST_ACCOUNT_CONNECTION_TIMESTAMP, $timestamp );
```

Comment:
I think the intention was probably to use `update_option` based on this comment:
> If the user changes tag settings this will get reset which makes sense because the connection parameters have been changed. ofc this will also get reset on a full reconnection.

Reply:
Feature removed.